server/maintenance_requests.go

```
@@ -1,34 +1,32 @@
 // This file provides NewApp() with functions for maintenance requests
 package main
 
-import(
-	"github.com/gin-gonic/gin"
-	"github.com/a2fumn2022/gpmn-nsl/server/datastore"
-	"time"
-	"io/ioutil"
-	"net/http"
+import (
 	"encoding/json"
-)
-
-/*type CreateRequest struct {
-	Request string `json:"request"`
-	UserSubmitted int `json:"id"`
-	DateSubmitted time.Time `json:"date"`
-}*/
+	"io/ioutil"
 
-/* AllMaintenanceRequests returns all the maintenance_requests from the database
-*  AllMaintenanceRequests is a function of app */
-func (app *App) AllMaintenanceRequests(c *gin.Context) {
-	/* Get data using psql.AllMaintenanceRequests and pass in app.db for the
-	* connection */
+	"net/http"
+	"time"
 
-	/* Takes the data returned in the psql func and formats it to a valid format
-	* using the gin.H func. See if it you can just use a Struct */
+	"github.com/a2fumn2022/gpmn-nsl/server/datastore"
+	"github.com/gin-gonic/gin"
+)
 
-	/* Use the gin.context.JSON func to return a status and the data. See auth.go
-	* for an example */
+// getMaintenanceRequests returns all the maintenance_requests from the database
+func (app *App) getMaintenanceRequests(c *gin.Context) {
+	requests, err := app.db.AllMaintenanceRequests()
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error!, Check to make sure your DB table is spelt correctly": err.Error()})
+		return
+	}
+	if requests == nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"Possible error!": "Your Table is Empty"})
```

Let's return the database error directly instead of using a string. Use this: `"error": err.Error()`

---

doesn't seem to work when calling direct error for an empty table

---

because error will be nil in this case. I think for this we just want to return an empty array. It's not an error that there's no rows in the table.

---

@mikelu92 so we return a status ok and an empty array?